### PR TITLE
Round trip prop and maskAccountId

### PR DESCRIPTION
**Pull request by benkolera**

### CODE REVIEW

File: consumer-data-au-api-types/tests/WaargoRoundTrip.hs [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+import           Waargonaut
+import           Waargonaut.Decode
+    (CursorHistory, Decoder, ppCursorHistory, simpleDecode)
+import           Waargonaut.Decode.Error
+import           Waargonaut.Encode          (Encoder', simpleEncodeNoSpaces)
+
+roundTripProp :: (Show a, Eq a) => Gen a -> Decoder Identity a -> Encoder' a -> TestName -> TestTree
+roundTripProp gen d e tn = testProperty tn . property $ do
+  a <- forAll gen
+  (Right a) === decodeBs d (encodeBs e a)
```
Comment (nghamilton) [inline consumer-data-au-api-types/tests/WaargoRoundTrip.hs]: Hedgehog's `tripping` function is pretty cool for roundtrip tests, if you haven't seen it.

Comment (benkolera) [reply consumer-data-au-api-types/tests/WaargoRoundTrip.hs]: Oh cool!

Comment (ajmcmiddlin) [inline consumer-data-au-api-types/tests/WaargoRoundTrip.hs]: Would [`tripping`](https://hackage.haskell.org/package/hedgehog-0.6.1/docs/Hedgehog.html#v:tripping) be any use here? Looks like it gives you some specialised output on failure at least.

File: consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Banking/Common/Accounts.hs [hunk 2]
```diff
@@ -102,6 +106,17 @@ newtype MaskedAccountNumber =
...
+-- e.g: 62 1234-5678 => XX XXXX-5678
+maskAccountId :: AccountId -> MaskedAccountNumber
+maskAccountId (AccountId (AsciiString t)) = MaskedAccountNumber $
+  (T.map mask $ T.take nonMasked t) <> (T.takeEnd 4 t)
+  where
+    chars = T.length t
+    mask c = bool c 'X' (isNumber c)
+    nonMasked = chars - 4
+
 maskedAccountNumberDecoder :: Monad f => Decoder f MaskedAccountNumber
```
Comment (nghamilton) [inline consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Banking/Common/Accounts.hs]: Does this need a check to ensure a MaskedAccountNumber isn't created from an AccountId?

Comment (benkolera) [reply consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Banking/Common/Accounts.hs]: It probably doesn't hurt. I need to get James to clearly say what the masking rules are first. They are a bit unclear atm to do this.

**Pull request merged**